docs/contributor/adr/0010-geoblocking-draft.md

```diff
@@ -0,0 +1,387 @@
+- the Config Map containing IP range allow/block list downloaded from the SAP internal service is a Kyma internal thing and the implementation may change at any time, so no other module should use it
+
+Consequence: Config map capacity may be exceeded in future, which may require immediate attention.
+
+### ip-auth deployment configurability
+
+The Geoblocking operator main functionality is the deployment of the ip-auth service. The deployment requires multiple parameters (like Docker image, resources). These parameters may be either configurable in the custom resource or just hardcoded.
+
+Decision: Hardcode deployment details in the API Gateway module.
+
+Consequence: It won't be possible to change the deployment details. 
+
+### Propagation of ip-auth service issues
+
+Geoblocking controller is responsible for the reconciliation of the Geoblocking custom resource, in particular: creating deployment of ip-auth service, creation of Authorization Policy, doing some configuration checks, etc. It may report issues related to above actions in the Geoblocking custom resource status.
+
+However, some issues may be visible only in the ip-auth service. The good example is a problem with the communication with SAP internal service (like a wrong secret).
+
+There is no easy way to propagate such issues to the Geoblocking CR. This would require a dedicated mechanism - like a 'health' endpoint exposed by ip-auth containers, so it can be analyzed by the Geoblocking controller.
+
+Decision: Don't implement additional mechanisms and follow the 'standard' approach, so the controller reports issues in the CR only from its own layer, while the ip-auth service reports issues in its log or via liveness/readiness probes (like every other workload).
+
+Consequence: Geoblocking CR won't be responsible for presenting Geoblocking 'health'. Support or developers must be aware that they should also inspect ip-auth logs in case of issues.
+
+### Namespaces
+
+Geoblocking mechanism touches multiple resources in multiple areas:
+- Istio configuration
+- Ingress gateway
+- Gateways in general
+- Authorization Policy
+- Authorizer workload (service, deployment)
+- Configuration of SAP internal service
+- Custom IP range allow/block list
+- IP range allow/block list cache
+
+It is a mix-in of different scopes - those managed by Kyma, by Istio and by the end-users.
+
+There are multiple factors that need to be taken into consideration:
+- Permissions - the Operator needs more permissions to access multiple namespaces
+- Kyma resources (workloads) should run in kyma-system namespace
+- kyma-system namespace should not be used by the end-users
+- Ingress Gateway is created by default in the istio-system namespace
+- Users may influence Istio (via Gateway API) to create Ingress Gateway in different namespace
+- Authorization Policy references Ingress Gateway via selector, so they should be in the same namespace
+- Gateway CR is supposed to be a singleton (for now)
```

Comment:
What Gateway CR? There are to many :D

Reply:
Good point, I probably confused it with Geoblocking CR